src/Bai7.cpp

```diff
@@ -0,0 +1,274 @@
+{
+    int cnt_child = countYounger(allPedestrians, 12);
+    auto impactToChildren = getImpact(cnt_child, 100, 0.01, 0.99);
+
+    int nPersonnelOfALKW = countALKW(allPedestrians);
+    auto impactToALKW = getImpact(nPersonnelOfALKW, 100, -0.29, 0.99);
+
+    int nPersonnelOfBFGMEN = countBFGMEN(allPedestrians);
+    auto impactToBFGMEN = multiplyByNegativeOne(getImpact(nPersonnelOfBFGMEN, 100, -0.29, 0.99));
+
+    int cnt_elder = countOlder(allPedestrians, 60);
+    auto impactToElder = multiplyByNegativeOne(getImpact(cnt_elder, 100, -0.29, 0.99));
+
+    int cnt_blinder = 0; // Giả sử không có người mù
+    auto impactToBlinder = multiplyByNegativeOne(getImpact(cnt_blinder, 100, -0.29, 0.99));
+
+    int nOthers = 0; // Đếm số người còn lại
+    for (const auto &p : allPedestrians)
+    {
+        if (!(p.getAge() < 12 || p.getAge() > 60 || p.getStart().getName() == "A" || p.getStart().getName() == "L" || p.getStart().getName() == "K" || p.getStart().getName() == "W" ||
+              p.getStart().getName() == "B" || p.getStart().getName() == "F" || p.getStart().getName() == "G" || p.getStart().getName() == "M" || p.getStart().getName() == "E" || p.getStart().getName() == "N"))
+        {
+            nOthers++;
+        }
+    }
+
+    auto impactToOthers = getImpact(nOthers, 100, -0.69, 0.99);
+
+    for (auto &p : allPedestrians)
+    {
+        vector<double> impactOfAGV(6, 0.0);
+        if (p.getAge() < 12)
+        {
+            for (int i = 0; i < 6; ++i)
+            {
+                impactOfAGV[i] += impactToChildren[i][0];
+            }
+            for (auto &row : impactToChildren)
+            {
+                row.erase(row.begin());
+            }
+        }
+        else if (p.getStart().getName() == "A" || p.getStart().getName() == "L" || p.getStart().getName() == "K" || p.getStart().getName() == "W")
+        {
+            for (int i = 0; i < 6; ++i)
+            {
+                impactOfAGV[i] += impactToALKW[i][0];
+            }
+            for (auto &row : impactToALKW)
+            {
+                row.erase(row.begin());
+            }
+        }
+        else if (p.getStart().getName() == "B" || p.getStart().getName() == "F" || p.getStart().getName() == "G" || p.getStart().getName() == "M" || p.getStart().getName() == "E" || p.getStart().getName() == "N")
+        {
+            for (int i = 0; i < 6; ++i)
+            {
+                impactOfAGV[i] += impactToBFGMEN[i][0];
+            }
+            for (auto &row : impactToBFGMEN)
+            {
+                row.erase(row.begin());
+            }
+        }
+        else if (p.getAge() > 60)
+        {
+            for (int i = 0; i < 6; ++i)
+            {
+                impactOfAGV[i] += impactToElder[i][0];
+            }
+            for (auto &row : impactToElder)
+            {
+                row.erase(row.begin());
+            }
+        }
+        else if (p.isBlind())
```

Comment:
Since p is `Pedestrian` and doesn't contain `Walkability` attribute, I think we can just add a `std::vector<Walkability> w` and check for `Walkability` based on `p.getID()`